src/SFA.DAS.AANHub.Api/Controllers/ApprenticesController.cs

```diff
@@ -30,39 +31,56 @@ public ApprenticesController(ILogger<ApprenticesController> logger, IMediator me
         [Produces("application/json")]
         [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
         [ProducesResponseType(typeof(string), StatusCodes.Status201Created)]
-        public async Task<IActionResult> CreateApprentice([FromHeader(Name = Constants.PostRequestHeaders.RequestedByUserHeader), Required] Guid? userId, CreateApprenticeModel request)
+        public async Task<IActionResult> CreateApprentice([FromHeader(Name = Constants.PostRequestHeaders.RequestedByUserHeader)] [Required] Guid? userId,
+            CreateApprenticeModel request)
         {
-            _logger.LogInformation("AAN Hub API: Received command to add apprentice by ApprenticeId: {apprenticeId} and UserId: {userId}", request.ApprenticeId, userId);
+            _logger.LogInformation("AAN Hub API: Received command to add apprentice by ApprenticeId: {apprenticeId} and UserId: {userId}",
+                request.ApprenticeId,
+                userId);
 
             CreateApprenticeMemberCommand command = request;
             command.RequestedByMemberId = userId;
 
             var response = await _mediator.Send(command);
-            return new CreatedAtActionResult(nameof(ApprenticesController.GetApprentice), "Apprentices", new { apprenticeid = request.ApprenticeId }, response);
+
+            return GetPostResponse(response,
+                new BaseRequestDetails
```

Comment:
I think having specific constructors in BaseRequest will help that way we can avoid repeating property names in each place.

Reply:
Controller name has been moved to constructor, and removed from ReferrerRouteDetails

Reply:
I need to improve on my comments. What I meant was that we add a constructor to ReferrerRouteDetails to take three arguments i.e., controller name, action and route parameters.

Reply:
Have moved the parameters to the constructor